### PR TITLE
[Merged by Bors] - feat: characterize "eventually" in a subtype

### PR DESCRIPTION
**Pull request by hrmacbeth**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

This is a pretty complicated proof of a simple statement, maybe there's some filter magic to improve it.

### CODE REVIEW

File: Mathlib/Topology/ContinuousOn.lean
Comment (ADedecker) [inline Mathlib/Topology/ContinuousOn.lean:522]:
```suggestion
  rw [← eventually_map, nhds_induced, nhdsWithin, map_comap_setCoe_val]
```

**Review by ADedecker — APPROVED** [review]
If you think it's useful, you could also add the more general
```lean
theorem Filter.eventually_comap' {f : β → α} (F : Filter α) (P : α → Prop) :
    (∀ᶠ x : β in comap f F, P (f x)) ↔ ∀ᶠ x in F ⊓ 𝓟 (range f), P x := by
  rw [← eventually_map, map_comap]
```
and deduce your version from it.

**Comment by ADedecker** [comment]
bors d+

**Comment by eric-wieser** [comment]
Is there a corresponding `frequently` result?

**Comment by ADedecker** [comment]
> Is there a corresponding `frequently` result?

Sure, it should follow almost immediately from the `eventually` result.

**Comment by eric-wieser** [comment]
I think the frequently result should be in this PR too, in that case

**Comment by ADedecker** [comment]
@hrmacbeth Do you mind if I take this over? I ended up needing it for Ascoli theorem.

**Comment by ADedecker** [comment]
I'm going to take this over, so I'll remove the "delegated" tag to get a second review from someone else.

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors r+

**Comment by hrmacbeth** [comment]
Sorry for the slow reply, thanks for taking over!

**Pull request closed without merging**
